Add seed argument to mesh sampling functions (#1682)

rowoflo opened the pull request:
This MR adds an additional argument to the `TriangleMesh::SamplePoints...` methods that is used to set the seed for the random number generator. The argument defaults to `-1`, which is a magic value that keeps the current functionality (the random number generator seed being set to a random number; thus, producing a new sequence which each trial). Any value besides `-1` will be used as the seed and will produce the same random number sequence for each trail. 

This MR addresses issue #1682 .

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/intel-isl/open3d/1683)
<!-- Reviewable:end -->

germanros1987 commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5303979&bid=17360599)

germanros1987 commented:
Thanks for the PR @rowoflo. Could you please have a look at Appveyor's report and try to fix it?

germanros1987 commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5303979&bid=17360599)

germanros1987 commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5303979&bid=17360599)

rowoflo commented:
Hi @germanros1987, I will need some assistance here. I am new to these CI tools and I am not what is causing the issue. It looks like Codacy is finding an issue with the `use_triangle_normal` argument (https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5303979). There should not be any changes to that argument.

yxlao commented:
Codacy is for our reference only, you can ignore this warning. However, the style check is failing for [Travis](https://travis-ci.org/github/intel-isl/Open3D/builds/671070486?utm_source=github_status&utm_medium=notification). Please follow http://www.open3d.org/docs/release/contribute.html#automated-style-checker and apply the style. You'll need version `5.0` of `clang-format` and `yapf==0.28.0`.

yxlao commented:
@rowoflo do you have any updates for the PR?

germanros1987 commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5303979&bid=17360599)

rowoflo commented:
@yxlao I addressed your comments and fixed the style errors based on clang-format version 9.

I tried installing clang-format5 using the instructions [here](http://www.open3d.org/docs/release/contribute.html#macos), but it just hangs forever trying to clone the repo. I left it for a couple of hours without any success.

germanros1987 commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5303979&bid=17360599)
